src/libpmemfile-posix/layout.h

```diff
@@ -161,28 +161,58 @@ struct pmemfile_inode {
 	 */
 	uint32_t suspended_references;
 
+	uint8_t padding1[48];
+
+	/* ---- cacheline boundary ----- */
```

Comment:
If this is expected, it would be even nicer to have `alignas(CL_SIZE)` set for these fields -- or actually a GNU attribute equivalent that works pre-C11. I checked `__attribute__ ((aligned (64)))`, works as expected. This way, the compiler knows about the alignmet as well (can use the alignment-expecting versions of some AVX2 instructions ), and there is no need to manually adding `padding[48]` anymore.

Reply:
I just realized that pmemobj 1.3 gives us only 16-byte aligned blocks (while 1.2 and master with headerless allocation classes gives us 64 byte aligned blocks), so:
- as long as we support nvml 1.3 we can't use this annotation
- this comment is not completely right

However we don't rely on cacheline boundaries anywhere (it's just an optimization), so these comments only document why those paddings are there.